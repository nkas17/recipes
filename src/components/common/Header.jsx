import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import LoadingDots from './LoadingDots';

/**
 * Common Header component
 */
const Header = ({ loading, navigation, authenticated }) => (
	<header>
		<h1 className="page-header">come and eat!</h1>
		{navigation && (
			<nav>
				<Link to="/recipe">recipes</Link>|<Link to="/about">about</Link>|
				<Link to="/login">log&nbsp;{authenticated ? 'out' : 'in'}</Link>
				{loading && <LoadingDots interval={200} dots={10} />}
			</nav>
		)}
	</header>
);

Header.defaultProps = {
	navigation: true,
};

Header.propTypes = {
	loading: PropTypes.bool.isRequired,
	navigation: PropTypes.bool,
	authenticated: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
	loading: state.numAjaxCallsInProgress > 0,
	authenticated: state.user.authenticated,
});

export default connect(mapStateToProps)(Header);
export { Header as HeaderTest, mapStateToProps };
