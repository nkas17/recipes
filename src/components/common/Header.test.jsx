import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HeaderTest, mapStateToProps } from './Header';

describe('Header renders', () => {
	it('correctly without loading', () => {
		const tree = shallow(<HeaderTest loading={false} authenticated={false} />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('correctly when authenticated', () => {
		const tree = shallow(<HeaderTest loading={false} authenticated />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('correctly while loading', () => {
		const tree = shallow(<HeaderTest loading authenticated={false} />);
		expect(toJson(tree)).toMatchSnapshot();
	});
});

describe('Header mapStateToProps runs', () => {
	const state = {
		numAjaxCallsInProgress: 0,
		user: { authenticated: false },
	};

	expect(mapStateToProps(state)).toStrictEqual({
		loading: false,
		authenticated: false,
	});
});
