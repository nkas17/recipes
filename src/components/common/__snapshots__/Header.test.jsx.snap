// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly while loading 1`] = `
<header>
  <h1
    className="page-header"
  >
    come and eat!
  </h1>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     | 
    <a
      href="/recipe"
      onClick={[Function]}
    >
      recipes
    </a>
     | 
    <a
      href="/mealList"
      onClick={[Function]}
    >
      meals
    </a>
     | 
    <a
      href="/groceryList"
      onClick={[Function]}
    >
      groceries
    </a>
     | 
    <a
      href="/about"
      onClick={[Function]}
    >
      about
    </a>
  </nav>
</header>
`;

exports[`Header renders correctly without loading 1`] = `
<header>
  <h1
    className="page-header"
  >
    come and eat!
  </h1>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     | 
    <a
      href="/recipe"
      onClick={[Function]}
    >
      recipes
    </a>
     | 
    <a
      href="/mealList"
      onClick={[Function]}
    >
      meals
    </a>
     | 
    <a
      href="/groceryList"
      onClick={[Function]}
    >
      groceries
    </a>
     | 
    <a
      href="/about"
      onClick={[Function]}
    >
      about
    </a>
  </nav>
</header>
`;
